portal/static/js/src/profile.js

```diff
@@ -2771,12 +2768,38 @@ export default (function() {
                     $(".put-roles-error").html("A role must be selected.");
                     return false;
                 }
+                var isChecked = $(event.target).is(":checked");
+                var changedRole = event.target.value;
+                var roles = [];
+                var self = this;
                 $(".put-roles-error").html("");
-                this.modules.tnthAjax.putRoles(this.subjectId, {"roles": roles}, $(event.target));
-                /*
-                 * refresh user roles list since it has been uploaded
-                 */
-                this.initUserRoles({clearCache: true});
+                $("#rolesGroup").addClass("loading");
+                this.modules.tnthAjax.getRoles(this.subjectId, function(data) {
+                    if (data.roles) {
+                        var dataRoles = data.roles.map(function(role) {
+                            return {
+                                name: role.name
+                            };
+                        });
+                        if (!isChecked) {
+                            //removed from existing role list
+                            dataRoles = dataRoles.filter(function(role){
+                                return role.name !== changedRole;
+                            });
+                            roles = dataRoles;
+                        } else {
+                            //combine checked role with existing roles
+                            roles = [{name: changedRole}, ...dataRoles];
```

Comment:
Does this array of combined roles need to be de-duplicated?

Reply:
thank you, Ivan, for catching this!  I did just fix it.